solr/core/src/java/org/apache/solr/security/PKIAuthenticationPlugin.java

```diff
@@ -347,22 +348,27 @@ PublicKey fetchPublicKeyFromRemote(String nodename) {
     HttpEntity entity = null;
     try {
       String uri = url + PublicKeyHandler.PATH + "?wt=json&omitHeader=true";
+      ModifiableSolrParams solrParams = new ModifiableSolrParams();
+      solrParams.add("wt", "json");
+      solrParams.add("omitHeader", "true");
+
+      GenericSolrRequest request =
+          new GenericSolrRequest(SolrRequest.METHOD.GET, PublicKeyHandler.PATH, solrParams);
+
       log.debug("Fetching fresh public key from: {}", uri);
-      HttpResponse rsp =
-          cores
-              .getUpdateShardHandler()
-              .getDefaultHttpClient()
-              .execute(new HttpGet(uri), HttpClientUtil.createNewHttpClientRequestContext());
-      entity = rsp.getEntity();
-      byte[] bytes = EntityUtils.toByteArray(entity);
-      Map<?, ?> m = (Map<?, ?>) Utils.fromJSON(bytes);
-      String key = (String) m.get("key");
-      if (key == null) {
-        log.error("No key available from {}{}", url, PublicKeyHandler.PATH);
-        return null;
-      } else {
-        log.info("New key obtained from node={}, key={}", nodename, key);
+
+      String key;
+      try (Http2SolrClient solrClient = new Http2SolrClient.Builder(url).build()) {
```

Comment:
why create a new client when you can use the default client?

Reply:
I was setting up the base path and initially used the `SolrRequest.setBasePath` method, which is now deprecated. I then tried to find a way to set the base path without using the deprecated API and remembered that we can recreate the builder with a URL to handle it. However, I know the consensus was to avoid repeatedly creating Http2SolrClient. Is there another way I can set the base path?

Reply:
I deprecated setBasePath but should have better clarified that there is no substitute yet and thus for code inside Solr (what we're talking about here) definitely should continue to use it.

Reply:
Wait, looks like I did not passed the Http2SolrClient to this builder.  Hmmmmm, I need to pass that here.